src/main/fc/rc_modes.c

```diff
@@ -177,3 +173,23 @@ void removeModeActivationCondition(const boxId_e modeId)
         }
     }
 }
+
+// Build the list of used modeActivationConditions indices
+// We can then use this to speed up processing by only evaluating used conditions
+void analyzeModeActivationConditions(void)
+{
+    modeActivationCondition_t emptyMac;
+    memset(&emptyMac, 0, sizeof(emptyMac));
+
+    activeMacCount = 0;
+    activeLinkedMacCount = 0;
+
+    for (uint8_t i = 0; i < MAX_MODE_ACTIVATION_CONDITION_COUNT; i++) {
+        const modeActivationCondition_t *mac = modeActivationConditions(i);
+        if (mac->linkedTo) {
+            activeLinkedMacArray[activeLinkedMacCount++] = i;
+        } else if (memcmp(mac, &emptyMac, sizeof(emptyMac))) {
```

Comment:
it would be nice to move empty MAC detection into separate function ..

Reply:
@ledvinap Good idea. I implemented this with a `isModeActivationConditionConfigured()` function. This also replaces configured/empty logic in `cli.c` as well as just provide general capability to easily check the state of a MAC.